### PR TITLE
Add support for playbook adjacent collections

### PR DESCRIPTION
**Pull request by priyamsahoo**
The PR adds support for playbook adjacent collections.
Resolves: #123, https://github.com/ansible/vscode-ansible/issues/631

### CODE REVIEW

File: src/services/ansibleConfig.ts [hunk 1]
```diff
@@ -97,6 +98,28 @@ export class AnsibleConfig {
         );
       }
     }
+
+    // add support for playbook adjacent collections
+    const docSettings = await this.context.documentSettings.get("ansible");
+    if (docSettings.ansible.supportPlaybookAdjacentCollections) {
+      const workspaceUri = this.context.workspaceFolder.uri;
+      console.log("*** workspace -> ", workspaceUri);
```
Comment (ganeshrn) [inline src/services/ansibleConfig.ts:106]: Is this log required?

File: test/providers/completionProvider.test.ts [hunk 2]
```diff
@@ -547,6 +598,7 @@ function testHostValues(
 
 describe("doCompletion()", () => {
   const workspaceManager = createTestWorkspaceManager();
+
```
Comment (ganeshrn) [inline test/providers/completionProvider.test.ts:601]: This space can be removed

File: test/providers/hoverProvider.test.ts [hunk 3]
```diff
@@ -176,8 +176,43 @@ function testNoHover(context: WorkspaceFolderContext, textDoc: TextDocument) {
...
+        await context.docsLibrary,
+      );
+      expect(actualHover.contents["value"]).includes(doc);
+    });
+  });
+}
+
 describe("doHover()", () => {
   const workspaceManager = createTestWorkspaceManager();
+
```
Comment (ganeshrn) [inline test/providers/hoverProvider.test.ts:215]: Same as above

**Review by ganeshrn — APPROVED** [review]
Apart from minor formatting comments, the overall PR looks good to me.

File: src/services/ansibleConfig.ts (hunk 1 above)
Comment (priyamsahoo) [reply src/services/ansibleConfig.ts]: Thanks for pointing that out.

File: Taskfile.yml [hunk 4]
```diff
@@ -127,6 +127,9 @@ tasks:
     cmds:
       - bash -c '{{ .NODE_ENV }} nvm install 14'
       - task: test
+      # - >
+      #   source $VIRTUAL_ENV/bin/activate &&
+      #   bash -c '{{ .NODE_ENV }} npm run test'
```
Comment (ganeshrn) [inline Taskfile.yml:132]: Is the test run not required here?

File: docs/settings.md [hunk 5]
```diff
@@ -14,6 +14,11 @@ Toggle usage of fully qualified collection names (FQCN) when inserting module na
 _default value:
 `true`_
 
+- **ansible.ansible.supportPlaybookAdjacentCollections**:
```
Comment (ganeshrn) [inline docs/settings.md:17]: Is `support` required in the setting name. How about `playbookAdjacentCollections`

File: src/services/docsLibraryUtilsForPAC.ts [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+      if (module) {
+        if (!hitFqcn) {
+          hitFqcn = fqcn;
+        }
+        break; // find first
+      }
+    }
+  }
+
+  console.log("*** module -> ", module);
```
Comment (ganeshrn) [inline src/services/docsLibraryUtilsForPAC.ts:106]: The log can be removed

**Review by ganeshrn — APPROVED** [review]
Looks good, can merge after the minor comments are fixed.

File: docs/settings.md (hunk 5 above)
Comment (priyamsahoo) [reply docs/settings.md:17]: I think `playbookAdjacentCollections` does not describe what the setting is related to. I followed the naming pattern used for other settings. For example, the other setting under the ansible section is `useFullyQualifiedCollectionNames`. This uses `use` and is self-descriptive.

At the same time, I feel like adding `support` lengthens the setting title, which is something we should avoid.

Comment (ajinkyau) [reply docs/settings.md:17]: can we use `enablePlaybookAdjacentCollections`?

Comment (ganeshrn) [reply docs/settings.md:17]: How about usePlaybookAdjacentCollection?

Comment (priyamsahoo) [reply docs/settings.md:17]: In continuation of the previous naming pattern, finalize on `usePlaybookAdjacentCollections`.

**Review by ajinkyau — APPROVED** [review]

**Pull request merged**